week-10/10-tech-lab.adoc

```diff
@@ -0,0 +1,228 @@
+.
+.
+.
+
+ * Consider hardening SSH configuration [SSH-7408]
+    - Details  : AllowAgentForwarding (YES --> NO)
+      https://cisofy.com/controls/SSH-7408/
+
+  * Configure minimum password age in /etc/login.defs [AUTH-9286]
+      https://cisofy.com/controls/AUTH-9286/
+
+  * Configure maximum password age in /etc/login.defs [AUTH-9286]
+      https://cisofy.com/controls/AUTH-9286/
+....
+
+Let's fix the minium and maxium password age issue in login.defs.
+
+Open file /etc/login.defs using text editor:
+# sudo nano /etc/login.defs
+
+Setup (sample) values as follows:
+
+ PASS_MAX_DAYS 30
+ PASS_MIN_DAYS 1
+ PASS_WARN_AGE 7
+
+To restart sshd on ubuntu 16, use the command:
+
+ sudo systemctl restart sshd
+
+For earlier versions of Ubuntu, use:
+
+ sudo service ssh restart
+
+Between these two resources you should be able to fix the issue. If you have questions, post them to Slack. When you have fixed the issue, run the scan again and confirm the banner is no longer an issue.
+
+=== System hardening
+
+Now for the hard part. Find another vulnerability you can fix. To use class time most efficiently, you should post the vulnerabilities you are researching to Slack. Don't start on something someone else is doing.
+
+You should take some time and think about and/or research the risk any given vulnerability presents. Remember the CISSP terminology:
+
+image::4.10-security-taxonomy.png[]
+
+We will pause class and discuss interesting cases.
+
+TIP: If you search Google on the Lynis ID, e.g. `AUTH-9286`, you should be taken to the cisofy.com site (or perhaps other resources) which will get you started on resolving it.
+
+NOTE: Do not try to fix the banners issue. It only can be tested by logging directly into the server console (which in the case of a VM doesn't really exist), not via ssh.
+
+Be prepared to share you results, positive or negative. If you fix a vulnerability, be sure to document the steps so they could be scripted.
+
+After you have fixed one more lynis vulnerability, read http://www.sonatype.org/nexus/2015/11/13/did-you-wake-up-to-an-alert-about-the-java-deserialization-vulnerability and discuss with your neighbors. What is the National Vulnerability Database? Be prepared to discuss in class.
+
+
+== Permanently fixing one or more vulnerabilities (10+ extra credit points).
+
+First, you need to figure out the configuration change required to fix the vulnerability. It should be something that you can script.
+
+* something installed that doesn't need to be
+* something that should be there, that isn't
+* permissions that need to be tightened down.
+
+* 1 vulnerability fixed: 10 points
+* 2 : 15 points
+* 3 : 20 points.
+* 10 or more: 30 points
+
+In order to gain the extra credit:
+
+* Create a new VM and run the vulnerability scanner.
+* Save the output to a text file, named "initial-output.log"
+* Run the necessary commands to fix the vulnerability.
+* Put them into a shell script.
```

Comment:
I would like them to create a hardened Packer image for extra credit.

Reply:
I added the comment, you'll have to review and let me know if I worded it the way you want.